libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -925,6 +1070,102 @@ impl NodeAPI for Greenlight {
         }
     }
 
+    async fn max_amount_to_send(
+        &self,
+        payee_node_id: Option<Vec<u8>>,
+        max_hops: u32,
+        last_hop_hints: Option<&RouteHintHop>,
+    ) -> NodeResult<(Vec<MaxChannelAmount>, PaymentPath)> {
+        let mut client = self.get_node_client().await?;
+
+        // Consider the hints as part of the route. If there is a routing hint we will
+        // attmept to calculate the path untill the last hop in the hint and then add
+        // the last hop to the path.
+        let (last_node, max_hops) = match last_hop_hints {
+            Some(hop) => (hex::decode(&hop.src_node_id)?, max_hops - 1),
+            None => match payee_node_id.clone() {
+                Some(node_id) => (node_id, max_hops),
+                None => {
+                    return Err(NodeError::RouteNotFound(anyhow!(
+                        "no payee node id or last hop hints provided, cannot calculate max amount"
+                    )));
+                }
+            },
+        };
+
+        // fetch a route from greenlight
+        let route_response = client
+            .get_route(GetrouteRequest {
+                id: last_node.clone(),
+                amount_msat: Some(pb::cln::Amount { msat: 1_000 }),
+                riskfactor: 0,
+                cltv: None,
+                fromid: None,
+                fuzzpercent: Some(0),
+                exclude: vec![],
+                maxhops: Some(max_hops),
+            })
+            .await?
+            .into_inner();
+
+        if route_response.route.is_empty() {
+            return Err(NodeError::RouteNotFound(anyhow!(
+                "no route found to node: {}",
+                hex::encode(&last_node),
+            )));
+        }
+
+        info!(
+            "max_amount_to_send: route response = {:?}",
+            route_response.route
+        );
+
+        // convert the route to a payment path so we can calculate the amount to forward for each hop
+        let mut payment_path = self.build_payment_path(route_response.route).await?;
+
+        // Add the last hop hints (if any) to the route
+        if let Some(hint) = last_hop_hints {
+            payment_path.edges.extend(vec![PaymentPathEdge {
+                base_fee_msat: hint.fees_base_msat,
+                fee_per_millionth: hint.fees_proportional_millionths,
+                node_id: payee_node_id.unwrap_or_default(),
+                short_channel_id: format_short_channel_id(hint.short_channel_id),
+                channel_delay: hint.cltv_expiry_delta,
+            }])
+        }
+
+        info!("max_amount_to_send: route_hops = {:?}", payment_path.edges);
+
+        // We fetch the opened channels so can calculate max amount to send for each channel
+        let (_, mut opened_channels, _, _) =
+            Self::fetch_channels_and_balance(client.clone()).await?;
+        opened_channels.sort_by(|c1, c2| {
```

Comment:
I think this sort is no longer necessary, because it's sorted here: https://github.com/breez/breez-sdk/pull/466/files#diff-7466a53546c6cded460ef5502a9168b13cc930d94fdf7d0ea8166bb10c75fe7eR604

Reply:
In the new code we are not using `fetch_channels_and_balance` anymore so the sorting is needed